File: app/src/main/java/org/stepic/droid/store/structure/DbStructureSections.java
package org.stepic.droid.store.structure;

public final class DbStructureSections extends DBStructureBase {

    private static String[] mUsedColumns = null;

    public static final String SECTIONS = "sections";

    public static final class Column {
        public static final String ID = "_id";
        public static final String SECTION_ID = "section_id";
        public static final String COURSE = "course";
        public static final String UNITS = "units";
        public static final String POSITION = "position";
        public static final String PROGRESS = "section_progress";
        public static final String TITLE = "title";
        public static final String SLUG = "slug";
        public static final String BEGIN_DATE = "begin_date";
        public static final String END_DATE = "end_date";
        public static final String SOFT_DEADLINE = "soft_deadline";
        public static final String HARD_DEADLINE = "hard_deadline";
        public static final String GRADING_POLICY = "grading_policy";
        public static final String BEGIN_DATE_SOURCE = "begin_data_source";
        public static final String END_DATE_SOURCE = "end_data_source";
        public static final String SOFT_DEADLINE_SOURCE = "soft_deadline_source";
        public static final String HARD_DEADLINE_SOURCE = "hard_deadline_source";
        public static final String GRADING_POLICY_SOURCE = "grading_policy_source";
        public static final String IS_ACTIVE = "is_active";
        public static final String CREATE_DATE = "create_date";
        public static final String UPDATE_DATE = "update_date";
        public static final String IS_CACHED = "is_cached";
        public static final String IS_LOADING = "is_loading";
    }

    public static String[] getUsedColumns() {
        if (mUsedColumns == null) {
            mUsedColumns = new String[]{
                    Column.ID,
                    Column.SECTION_ID,
                    Column.TITLE,
                    Column.SLUG,
                    Column.IS_ACTIVE,
                    Column.BEGIN_DATE,
                    Column.SOFT_DEADLINE,
                    Column.HARD_DEADLINE,
                    Column.COURSE,
                    Column.POSITION,
                    Column.UNITS,
                    Column.IS_CACHED,
                    Column.IS_LOADING,
            };
        }
        return mUsedColumns;
    }
}

File: app/src/main/java/org/stepic/droid/storage/structure/DbStructureSections.kt
package org.stepic.droid.storage.structure

@Deprecated("Use cache.DbStructureSection")
object DbStructureSections {
    const val SECTIONS = "sections"

    object Column {
        @Deprecated("it is not used, use section_id instead")
        const val ID = "_id"
        const val SECTION_ID = "section_id"
        const val COURSE = "course"
        const val UNITS = "units"
        const val POSITION = "position"
        const val PROGRESS = "section_progress"
        const val TITLE = "title"
        const val SLUG = "slug"
        const val BEGIN_DATE = "begin_date"
        const val END_DATE = "end_date"
        const val SOFT_DEADLINE = "soft_deadline"
        const val HARD_DEADLINE = "hard_deadline"
        const val GRADING_POLICY = "grading_policy"
        const val BEGIN_DATE_SOURCE = "begin_data_source"
        const val END_DATE_SOURCE = "end_data_source"
        const val SOFT_DEADLINE_SOURCE = "soft_deadline_source"
        const val HARD_DEADLINE_SOURCE = "hard_deadline_source"
        const val GRADING_POLICY_SOURCE = "grading_policy_source"
        const val IS_ACTIVE = "is_active"
        const val CREATE_DATE = "create_date"
        const val UPDATE_DATE = "update_date"
        const val IS_CACHED = "is_cached"
        const val IS_LOADING = "is_loading"
        const val TEST_SECTION = "can_test_section"
        const val DISCOUNTING_POLICY = "discounting_policy"
        const val IS_EXAM = "is_exam"
        const val IS_REQUIREMENT_SATISFIED = "is_requirement_satisfied"
        const val REQUIRED_SECTION = "required_section"
        const val REQUIRED_PERCENT = "required_percent"
    }
}
